virttest/utils_disk.py

```diff
@@ -204,26 +205,22 @@ def get_linux_disks(session, partition=False):
     :return: the disks dict.
              e.g. {kname: [kname, size, type, serial, wwn]}
     """
-    list_disk_cmd = "lsblk -o KNAME,SIZE,TYPE,SERIAL,WWN"
-    get_disk_name_cmd = "lsblk -no pkname /dev/%s"
-    output = session.cmd_output(list_disk_cmd)
-    devs = output.splitlines()
     disks_dict = {}
-    part_dict = {}
-    for dev in devs:
-        dev = dev.split()
-        if "disk" in dev:
-            disks_dict[dev[0]] = dev
-        if "part" in dev:
-            part_dict[dev[0]] = dev
-    if partition:
-        disks_dict = dict(disks_dict, **part_dict)
-    else:
-        for part in part_dict.keys():
-            output = session.cmd_output(get_disk_name_cmd % part)
-            disk = output.splitlines()[0].strip()
-            if disk in disks_dict.keys():
-                disks_dict.pop(disk)
+    parent_disks = set()
+    block_info = session.cmd('ls /sys/dev/block -l | grep \'/pci\'')
+    for matched in re.finditer(r'/block/(\S+)/\S+\s^', block_info, re.M | re.I):
+        parent_disks.add(matched.group(1))
+    for matched in re.finditer(r'/block/(\S+)\s^', block_info, re.M | re.I):
+        knames = matched.group(1).split('/')
+        if partition is False and knames[0] in parent_disks:
+            continue
+        disks_dict[knames[-1]] = []
+        o = session.cmd('udevadm info -q all -n {0} && '
+                        'lsblk -o KNAME,SIZE | grep \'{0} \''.format(knames[-1]))
+        for parttern in (r'\w+\S+\s+(\d+.?\d+\w+)\s^', r'DEVTYPE=(\w+)\s^',
```

Comment:
pattern to get disk/ partition size is  not rigorous. In my rhel8.1.0 guest, I get return 
{'sda': ['20G', 'disk', None, None], 'sda1': [None, 'partition', None, None], 'sdb': [None, 'disk', None, None], 'sda2': ['001', 'partition', None, None]}